### DISCUSSION TITLE
Nonexistent PR causing Renovate to no longer open pull requests

### DISCUSSION THREAD

**Discussion by xaevik** [renovatebot/renovate#24418] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

We have one repository where renovate is no longer opening pull requests for non-major dependency updates. According to the logs this is due to it looking for a PR that no longer exists (189) as we had to have GitHub remove it due to corruption.

As such we are having to manually run composer updates to keep `composer.lock` up to date. We are not sure if this is due to a caching mechanism in which now the cache is poisoned. We have already tried re-installing the app and waiting between 24 and 48 hours, but it has had no effect.

As this (what we feel) is an edge case on a private repository and contains non-trivial set of conditions that cannot be easily reproduced, a sample repository cannot be provided.

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Found PR #189
DEBUG: 0 PRs are currently open
DEBUG: PR concurrent limit remaining: 10
DEBUG: Calculated maximum PRs remaining this run: 2
DEBUG: PullRequests limit = 2
DEBUG: Calculating hourly PRs remaining
DEBUG: currentHourStart=2023-09-14T09:00:00.000+00:00
DEBUG: PR hourly limit remaining: 2
DEBUG: Calculating branchConcurrentLimit (10)
DEBUG: 0 already existing branches found: 
DEBUG: Branch concurrent limit remaining: 10
DEBUG: Calculated maximum branches remaining this run: 2
DEBUG: Branches limit = 2
DEBUG: syncBranchState() (branch="renovate/lock-file-maintenance")
DEBUG: getBranchPr(renovate/lock-file-maintenance) (branch="renovate/lock-file-maintenance")
DEBUG: findPr(renovate/lock-file-maintenance, undefined, open) (branch="renovate/lock-file-maintenance")
DEBUG: findPr(renovate/lock-file-maintenance, undefined, closed) (branch="renovate/lock-file-maintenance")
DEBUG: Found PR #189 (branch="renovate/lock-file-maintenance")
DEBUG: branchExists=false (branch="renovate/lock-file-maintenance")
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment]
The logs you provided don't fully match with the description you gave or show any blocking. 

Logs are incomplete, but also maybe the description can be clearer? Is one branch blocked or many?

**Comment by xaevik** [comment]
Hi @rarkins,

Understood, I'll try and articulate this in a bit more detail. Originally Renovate had opened a pull request, in this case PR 189, to handle non-major updates. The PR was inadvertently modified to attempt to rectify an issue where Renovate is running a newer version of composer than a dependency is compatible with (it is locked with the LTS release of composer). 

When this happened renovate deemed the PR was edited and would refuse to rebase (we attempted several times) and we eventually merged the PR manually and deleted the branch. After this occurred, renovate simply stopped opening new PRs for non-major dependency updates while continuing to reference the now closed PR.

We then reached out to GitHub support and asked them to delete the PR as a way of hopefully kick starting renovate to start opening new pull requests and no longer find the aforementioned PR. As you can see from the narrow log snippet it still is referencing the now deleted PR which we believe is blocking renovate from continuing. 

Our current `renovate.json` is below:

```json
{
    "extends": [
        "config:base",
        ":disableDependencyDashboard",
        ":gitSignOff",
        ":preserveSemverRanges"
    ],
    "automerge": true,
    "automergeType": "pr",
    "platformAutomerge": true,
    "ignoreDeps": [
        "magento/magento-cloud-metapackage",
        "magento/product-enterprise-edition",
        "tedivm/jshrink",
        "phpunit/phpunit"
    ],
    "ignorePaths": [
        "app/code/**/composer.json",
        "app/design/**"
    ],
    "ignoreTests": true,
    "rangeStrategy": "update-lockfile",
    "lockFileMaintenance": {
        "enabled": true,
        "extends": [
            "schedule:daily"
        ]
    },
    "hostRules": [
        {
            "matchHost": "repo.packagist.com",
            "hostType": "packagist",
            "username": "token",
            "encrypted": {
                "password": "[redacted]"
            }
        }
    ],
    "packageRules": [
        {
            "matchPackagePatterns": [
                "*"
            ],
            "matchUpdateTypes": [
                "minor",
                "patch",
                "digest"
            ],
            "groupName": "all non-major dependencies",
            "groupSlug": "all-minor-patch",
            "labels": [
                "dependencies"
            ]
        },
        {
            "matchPackagePatterns": [
                "*"
            ],
            "matchUpdateTypes": [
                "major"
            ],
            "labels": [
                "dependencies",
                "breaking"
            ],
            "automerge": false,
            "assigneesFromCodeOwners": true
        }
    ]
}
```

**Reply by rarkins** [reply]
I recommend you remove automerge=true in your root config. It has the same effect as setting separateMajorMinor=false and is one reason why you may not be seeing non-major updates as much as you should. I also strongly recommend against automerging major updates.

Back to the topic. You are saying "non-major" updates, but I think it would be more exact to say "Lock file maintenance PRs are blocked by an existing, now-deleted lock file maintenance PR"?

Assuming so, the logs you gave are incomplete and don't show any blocking. Please include all log messages which include `(branch="renovate/lock-file-maintenance")`

**Comment by xaevik** [comment]
I am not sure I understand your explanation about the `automerge` settings, we do specifically turn off `automerge` for the major group and this configuration has worked on all our repositories. In any case, we'll make the adjustment.

You are correct in the title clarification, which was my fault for not being more concise.

Here is the requested log snippet:

```
DEBUG: detectSemanticCommits()
DEBUG: semanticCommits: returning "enabled" from cache
DEBUG: processRepo()
DEBUG: Processing 1 branch: renovate/lock-file-maintenance
DEBUG: Calculating hourly PRs remaining
DEBUG: currentHourStart=2023-09-14T09:00:00.000+00:00
DEBUG: PR hourly limit remaining: 2
DEBUG: Calculating prConcurrentLimit (10)
DEBUG: getBranchPr(renovate/lock-file-maintenance)
DEBUG: findPr(renovate/lock-file-maintenance, undefined, open)
DEBUG: findPr(renovate/lock-file-maintenance, undefined, closed)
DEBUG: Found PR #189
DEBUG: 0 PRs are currently open
DEBUG: PR concurrent limit remaining: 10
DEBUG: Calculated maximum PRs remaining this run: 2
DEBUG: PullRequests limit = 2
DEBUG: Calculating hourly PRs remaining
DEBUG: currentHourStart=2023-09-14T09:00:00.000+00:00
DEBUG: PR hourly limit remaining: 2
DEBUG: Calculating branchConcurrentLimit (10)
DEBUG: 0 already existing branches found: 
DEBUG: Branch concurrent limit remaining: 10
DEBUG: Calculated maximum branches remaining this run: 2
DEBUG: Branches limit = 2
DEBUG: syncBranchState() (branch="renovate/lock-file-maintenance")
DEBUG: getBranchPr(renovate/lock-file-maintenance) (branch="renovate/lock-file-maintenance")
DEBUG: findPr(renovate/lock-file-maintenance, undefined, open) (branch="renovate/lock-file-maintenance")
DEBUG: findPr(renovate/lock-file-maintenance, undefined, closed) (branch="renovate/lock-file-maintenance")
DEBUG: Found PR #189 (branch="renovate/lock-file-maintenance")
DEBUG: branchExists=false (branch="renovate/lock-file-maintenance")
DEBUG: dependencyDashboardCheck=undefined (branch="renovate/lock-file-maintenance")
DEBUG: recreateClosed is true. No need to check for closed PR. (branch="renovate/lock-file-maintenance")
DEBUG: Checking schedule(before 4am, null) (branch="renovate/lock-file-maintenance")
DEBUG: Checking 1 schedule(s) (branch="renovate/lock-file-maintenance")
DEBUG: Checking schedule "before 4am" (branch="renovate/lock-file-maintenance")
{
  "parsedSchedule": {
    "schedules": [
      {
        "t_b": [
          14400
        ]
      }
    ],
    "exceptions": [],
    "error": -1
  }
}

DEBUG: Package not scheduled (branch="renovate/lock-file-maintenance")
DEBUG: Skipping branch creation as not within schedule (branch="renovate/lock-file-maintenance")
```

**Note:** No such branch exists on the origin either, we have checked. We've also adjusted different schedules, again this is only happening on this one specific repository, we have 28 others and they are not experiencing the same issue.

**Reply by rarkins** [reply, marked as answer]
It says `Skipping branch creation as not within schedule`

**Reply by xaevik** [reply]
Going by the logs on all the previous executions, it's never hitting the scheduling. It always hits 3 hours before or 1 hour after the scheduled time. We can try and set explicit scheduling but it seems odd that this is only occurring on this one and only repository.

**Reply by rarkins** [reply]
If you used dependency dashboard then you could force such a PR on demand at any time

**Reply by xaevik** [reply]
We got it operating again, the configuration was incorrect (how it was working before is probably an oddity). We moved the `automerge` configuration to `lockFileMaintenance` and out of root and adjusted the scheduling:

```json
    "lockFileMaintenance": {
        "enabled": true,
        "automerge": true,
        "automergeType": "pr",
        "platformAutomerge": true,
        "schedule": [
          "after 4:00am"
        ]
    },
```

It still fails due to:

```
INFO: Composer requirements cannot be resolved
```

However, this is caused by a transitive dependency that has a dependency on [laminas/laminas-dependency-plugin](https://github.com/laminas/laminas-dependency-plugin) and they have locked that package as only being compatible with Composer 2.2. We're working with the maintainer to try and remove that dependency.
